controllers/cleanup.go

```diff
@@ -88,7 +88,7 @@ func (r *SearchReconciler) deleteLegacyServiceMonitorSetup(instance *searchv1alp
 	if err = r.Delete(r.context, r.MetricsRoleBinding(instance)); err != nil && !errors.IsNotFound(err) {
 		log.Error(err, "Failed to remove RoleBinding")
 	}
-	if err == nil {
+	if err == nil || (err != nil && errors.IsNotFound(err)) {
 		log.Info("Deleted legacy ServiceMonitor Setup from openshift-monitoring namespace")
```

Comment:
Should I remove this logging as error can happen in a previous step and might not reach here as others may succeed?

Reply:
Right, `err` will only have the result of the last delete operation. We can log the result of each resource because it will only happen once.

Reply:
Removed it.